src/storage/hummock_sdk/src/sstable_info_ref.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::{Arc, Weak};
+
+use risingwave_pb::hummock::{PbBloomFilterType, PbHummockVersion, PbSstableInfo};
+use tokio::task_local;
+
+use crate::change_log::EpochNewChangeLogCommon;
+use crate::key_range::KeyRange;
+use crate::sstable_info::SstableInfo;
+use crate::version::{HummockVersion, HummockVersionCommon, HummockVersionDeltaCommon};
+use crate::HummockSstableId;
+
+pub type SstableInfoType = SstableInfoRef;
+pub type EpochNewChangeLogType = EpochNewChangeLogCommon<SstableInfoType>;
+pub type HummockVersionType = HummockVersionCommon<SstableInfoType>;
+pub type HummockVersionDeltaType = HummockVersionDeltaCommon<SstableInfoType>;
+
+pub trait SstableInfoReader {
+    fn object_id(&self) -> u64;
+    fn sst_id(&self) -> u64;
+    fn key_range(&self) -> &KeyRange;
+    fn file_size(&self) -> u64;
+    fn table_ids(&self) -> &[u32];
+    fn meta_offset(&self) -> u64;
+    fn stale_key_count(&self) -> u64;
+    fn total_key_count(&self) -> u64;
+    fn min_epoch(&self) -> u64;
+    fn max_epoch(&self) -> u64;
+    fn uncompressed_file_size(&self) -> u64;
+    fn range_tombstone_count(&self) -> u64;
+    fn bloom_filter_kind(&self) -> PbBloomFilterType;
+    fn sst_size(&self) -> u64;
+}
+
+pub trait SstableInfoWriter {
+    fn set_sst_id(&mut self, v: u64);
+    fn set_sst_size(&mut self, v: u64);
+    fn set_table_ids(&mut self, v: Vec<u32>);
+}
+
+task_local! {
+    pub static SSTABLE_INFO_CACHE: *mut SstableInfoCache;
+}
+
+#[derive(Debug, PartialEq, Clone, Default)]
+pub struct SstableInfoRef {
```

Comment:
As we have implemented our own `SstableInfo` struct, can we just make the implementation of the clone of `SstableInfo` a shallow clone instead of explicitly introducing a struct wrapping it with arc? In this way, most of the changes in this PR can be avoided.